content/chainguard/chainguard-images/comparing-images.md

```diff
@@ -0,0 +1,145 @@
+---
+title: "How To Compare Chainguard Images with chainctl"
+linktitle: "Comparing Images"
+type: "article"
+description: "An overview of how to use the chainctl images diff command to compare two Chainguard Images."
+date: 2023-08-30T11:07:52+02:00
+lastmod: 2023-08-30T11:07:52+02:00
+draft: false
+tags: ["Chainguard Images", "Product"]
+images: []
+menu:
+  docs:
+    parent: "chainguard-images"
+weight: 400
+toc: true
+---
+
+Say that you're curious about the difference between two Chainguard Images. For example, you might want to know if there are any significant differences between yesterday's build and today's; or perhaps you want to know if any CVEs are present in a newer version of a custom Image.
+
+[`chainctl`](/chainguard/chainctl/) — Chainguard's command line interface tool — allows you to directly compare two Chainguard Images with its `images diff` feature. This guide outlines how to use the Image diffing feature and highlights a few potential use cases for it.
+
+
+## Prerequisites
+
+In order to use the `chainctl images diff` subcommand, you'll need to have a few tools installed.
+
+* You'll need `chainctl` installed on your local machine. Follow our guide on [How to Install chainctl](/chainguard/chainguard-enforce/how-to-install-chainctl/) to set this up.
+* Next, ensure you have Cosign installed. Our guide on [How to Install Cosign](/open-source/sigstore/cosign/how-to-install-cosign/) outlines several methods for installing Cosign.
+* You'll also need Grype installed on your local machine, as `chainctl` uses this to scan the images when performing the diff. Follow the installation instructions for your operating system on the [Grype project GitHub repository](https://github.com/anchore/grype#installation).
+
+
+## Using `chainctl images diff`
+
+The `chainctl images diff` subcommand accepts the names of two Chainguard Images as arguments and uses Grype to perform a vulnerability scan on each of them. It then retrieves both Images' SBOM information and outputs the difference between the two along with the previously obtained Grype data.
+
+The `diff` subcommand follows this general syntax.
+
+```sh
+chainctl images diff $FROM_IMAGE $TO_IMAGE
+```
+
+As an example, try comparing the `latest` public `go` Chainguard Image with its `latest-dev` version.
+
+```sh
+chainctl images diff cgr.dev/chainguard/go:latest cgr.dev/chainguard/go:latest-dev | jq
+```
+
+This will return output like the following.
+
+```
+Fetching vulnerabilities for cgr.dev/chainguard/go@sha256:6fee3fff87854aa6e4762c7998c127436a68b09877f9c1010deca35e0f1e27bc
```

Comment:
There's a chance the format of the output will change soon, maybe we could just put in a note at the top of this doc that this feature is still in progress and subject to change? Thank you!